challenge_3/c/scottsquatch/src/main.c

```diff
@@ -0,0 +1,94 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+/** Constants **/
+#define MODE_READ  "r"
+#define NUM_ARGS 3
+#define FILE_NAME_ARG 1
+#define ARRAY_SIZE_ARG 2
+
+/** Function Prototypes **/
+int int_compare(const void*, const void*);
+void print_array(int*, int);
+int get_majoriy_element(int*, int);
+int* read_array(char*,int);
+
+int main(int argc, char* argv[])
+{
+  // Get parameters from input
+  if (argc != NUM_ARGS)
+  {
+    // Incorrect number of parameters
+    printf("Incorrect Number of parameters, expected %d but got %d\n",
+      NUM_ARGS - 1, argc - 1);
+    printf("Usage: ./majority_element <array file> <size of array in file>\n");
```

Comment:
A `main` function that lacks an explicit return on the main branch implicitly returns 0. I'm not sure if that's in the standard or just a thing most compilers do, but `return 0;` is inserted at the end of `main()` in at least GCC and Clang.